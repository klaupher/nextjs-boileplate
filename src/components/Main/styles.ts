'use client'
import styled from "styled-components";
import { TbSunFilled, TbMoonFilled} from 'react-icons/tb';

export const Wrapper = styled.div`
  background-color: var(--bg);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  padding: 2rem;

  @media (min-width: 40em) {
      padding: 6rem;
  }
`

export const Logo = styled.img`
  width: 25rem;
  margin: 2rem;
`
export const Title = styled.h1`
  font-size: 2.5rem;
  color: var(--headingcolor);
`
export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  color: var(--headingcolor);
`
export const Illustation = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%)
`

export const ThemeToggleContainer = styled.div`
  display: flex;
  align-self: flex-end
`

export const SunIcon = styled(TbSunFilled)`
  cursor: pointer
`

export const MoonIcon = styled(TbMoonFilled)`
  cursor: pointer
`
