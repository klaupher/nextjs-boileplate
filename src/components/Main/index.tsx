import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src='/img/logo.svg' alt='Imagem de um átomo e o texto React Avançado ao lado'/>
    <S.Title>React Avançado</S.Title>
    <S.Description>Typescript, ReactJS, NextJS e StyledComponents</S.Description>
    <S.Illustation src='/img/hero-illustration.svg' alt='Desenvolvedor de costas codificando' />
  </S.Wrapper>
);

export default Main;
