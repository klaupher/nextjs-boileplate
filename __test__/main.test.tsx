import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { describe } from 'node:test';
import Main from '@/app/Components/Main';

describe('<Main />', () => {
  it('shoudl render the heading', () => {
    render(<Main />);
    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument();
  });

  it('should render the color specific', () => {
    const {container} = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color':'#06092b'})
  })
});
