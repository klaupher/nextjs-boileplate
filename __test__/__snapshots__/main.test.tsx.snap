// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoudl render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e o texto React Avançado ao lado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e StyledComponents
  </h2>
  <img
    alt="Desenvolvedor de costas codificando"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;

exports[`should render the color specific 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e o texto React Avançado ao lado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e StyledComponents
  </h2>
  <img
    alt="Desenvolvedor de costas codificando"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
